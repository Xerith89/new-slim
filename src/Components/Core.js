import React from 'react'
import PropTypes from 'prop-types'

export function Spinner(props) {
    return (
        <div className="d-flex justify-content-center m-5" data-testid='spinner'>
            <div style={props.style} className="spinner-border" role="status">
            </div>
      </div>
    )
}

export function Button(props) {
    return (
        <div>
            <button type={props.type} className={props.className} name={props.name} onClick={props.onClick} style={props.style}>{props.display}</button>
        </div>
    )
}

export function Card(props) {
    return (
        <div>
        <div className={`card `+props.className} style={props.style}>
            <div>{props.children ? props.children : null}</div>
        </div>
    </div>
    )
}

export function CardTitle(props) {
    return (
        <h2 style={props.style} className={`card-header `+props.className}>{props.title}</h2>
    )
}

export function CardBody(props) {
    return (
        <div style={props.style} className={`card-body `+props.className}>
            {props.children ? props.children : null}
        </div>
    )
}

Button.propTypes = {
    style: PropTypes.object,
    display: PropTypes.any,
    className: PropTypes.string,
    onClick: PropTypes.func,
    name: PropTypes.string,
    type: PropTypes.string
}

Card.propTypes = {
    children: PropTypes.any,
    style: PropTypes.object,
    className: PropTypes.string,
}

CardTitle.propTypes = {
    title: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
}

CardBody.propTypes = {
    className: PropTypes.string,
    style: PropTypes.object,
    children: PropTypes.any
}


Spinner.propTypes = {
    style: PropTypes.object,
    className: PropTypes.string
}


