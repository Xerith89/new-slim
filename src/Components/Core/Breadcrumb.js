import React from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTachometerAlt} from '@fortawesome/free-solid-svg-icons'

export default function Breadcrumb(props) {
    return (
        <div className="container mt-3">
            <h1 className="text-center"><FontAwesomeIcon icon={faTachometerAlt}/> {props.title}</h1>
        </div>
    )
}

Breadcrumb.propTypes = {
    title: PropTypes.string
}
