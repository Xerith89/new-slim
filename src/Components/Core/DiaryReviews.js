import React, { Component } from 'react'

export default class DiaryReviews extends Component {

    render() {

        const style = {
            width: "100%"
        }
        return (
            <div>
                <div className="col-md-12">
                    <div id="overview" className="card" style={style}>
                        <h5 className="card-header main-color-bg">Search</h5>
                        <div className="card-body">
                            <div className="card-deck">
                                <div className="card">

                                </div>
                            </div>
                        </div>
                    </div> 
                </div>
            </div>
        )
    }
}
