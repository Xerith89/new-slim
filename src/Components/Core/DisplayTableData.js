import React from 'react'
import PropTypes from 'prop-types'

export default function DisplayTableData(props) {
    return (
       <React.Fragment>
        {props.finalData && props.finalData.map((task) => {
            return (
            <tr key={task.id}>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.name}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.reference}</a></td>
                <td><a href={"/"+task.type+"/"+task.id} style={{display: 'block'}}>{task.type}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.assigned}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.priority}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.due}</a></td>
            </tr>
           )
        })}
    </React.Fragment>     
    )
}

DisplayTableData.propTypes = {
    taskList: PropTypes.arrayOf(Object)
}



