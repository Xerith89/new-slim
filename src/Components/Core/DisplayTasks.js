import React from 'react'

export default function DisplayTasks(props) {
    return (
        <tbody data-testid='tasks'>
        {props.taskList && props.taskList.map((task) => {
            return (
            <tr key={task.id}>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.taskName}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.claimSpecNo}</a></td>
                <td><a href={"/"+task.type+"/"+task.id} style={{display: 'block'}}>{task.type}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.assigned}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.priority}</a></td>
                <td><a href={"/"+task.type/task.id} style={{display: 'block'}}>{task.dueDate}</a></td>
            </tr>
           )
        })}
    </tbody>     
    )
}


