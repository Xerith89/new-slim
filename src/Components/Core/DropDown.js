import React from 'react'

export default function DropDown() {
    return (
        <div data-testid="menu">
            <ul className="list-group dropdown-menu dropdown-menu-right">
                <li className="dropdown-item"><a href="/settings">Settings</a></li>
                <div className="dropdown-divider"></div>
                <li className="dropdown-item"><a href="/logout">Logout</a></li>
            </ul>
        </div>
    )
}