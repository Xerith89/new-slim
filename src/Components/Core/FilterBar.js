import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {filterByString} from '../Util/Filter'
import Pagination from './Pagination';

export default class FilterBar extends Component {

    constructor(props){
        super(props)

        this.state = this.getInitialState();
    }

    getInitialState = () => {
        const initialState = {
            filteredData:  this.props.data,
        }
        this.props.filterOptions.forEach((e) => {
            initialState[e.header] = ''
        })
        
        return initialState;
    }

    clearfilter = () => {
        this.setState(this.getInitialState());
    }

    handleChange = (event) => {
        let refiltered = this.props.data;
        //We're taking off a filter so recursively check all the others that are set
        if (event.target.value === '') {
            this.props.filterOptions.forEach((dropdown) => {
                if (dropdown.header !== event.target.name) {
                    refiltered = filterByString(refiltered, dropdown.header, this.state[dropdown.header]);
                }                
            });
           this.setState({
               filteredData: refiltered,
               [event.target.name]: event.target.value
           });
           //Setting a filter
        } else {
            this.props.filterOptions.forEach((dropdown) => {
                if (this.state[dropdown.header] !== '' && event.target.name !== dropdown.header) {
                    refiltered = filterByString(refiltered,dropdown.header,this.state[dropdown.header]);
                } else if(event.target.name === dropdown.header) {
                    refiltered = filterByString(refiltered,dropdown.header,event.target.value);
                }
            });
            this.setState({
                filteredData: refiltered,
                [event.target.name] : event.target.value
            });
        }
        
    }
   
    render() {
        return (
            <div>
                 <div id="overview" className="card m-4" style={{width: 'auto'}} data-testid="filterBar">
                 <div className="form-group m-2">
                 {this.props.filterOptions && this.props.filterOptions.map((element) => { 
                     return (<React.Fragment key={element.header}>
                                    <label className="mr-1 ml-2 mt-3 mb-2">{element.header}:</label>
                                    <select id={element.header} name={element.header} value={this.state[element.header]} onChange={this.handleChange} className="form-control-sm col-sm-2 mr-2">
                                    {element.body.map((option, j) => { 
                                    return( <option value={option} key={j}>{option}</option>)
                                    })}
                                    </select>
                            </React.Fragment>)
                            })}
                         <button onClick={this.clearfilter} className="btn btn-primary ml-3">Clear Filter</button>
                       </div>
                 </div>
                 <Pagination fetching={this.props.fetching} recordsPerPage={6} filteredData={this.state.filteredData} tableHeaderOptions={this.props.tableHeaderOptions}/>
            </div>
        )
    }
}

FilterBar.propTypes = {
    options: PropTypes.arrayOf(Object),
    fetchingTasks: PropTypes.bool,
    taskList: PropTypes.array
}
