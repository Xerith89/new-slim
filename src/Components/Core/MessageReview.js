import React, { Component } from 'react'

export default class MessageReview extends Component {

    render() {

        return (
            <div >
                <div className="mt-5">
                    <div id="overview" className="card" style={{width: 'auto'}}>
                        <h5 className="card-header text-center" style={{backgroundColor: "#ebf2fc"}}>Messages <span className="badge badge-dark">3</span></h5>
                        <div className="card-body">
                            <div className="form-group">
                            <select className="form-control-sm">
                                <option>My Messages</option>
                                <option>Team Messages</option>
                            </select>
                            </div>
                            <div className="m-3">
                                <table className="table table-striped table-hover">
                                    <thead>
                                        <tr>
                                        <th scope="col">Name</th>
                                        <th scope="col">Date</th>
                                        <th scope="col">Subject</th>
                                        <th scope="col"></th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        <tr>
                                        <td>Mark Otto</td>
                                        <td>12-04-19</td>
                                        <td>Callback</td>
                                        <td></td>
                                        </tr>
                                        <tr>
                                        <td>Jacob Thornton</td>
                                        <td>14-04-19</td>
                                        <td>New Claim</td>
                                        <td></td>
                                        </tr>
                                        <tr>
                                        <td>Larry</td>
                                        <td>Bird</td>
                                        <td>18-04-19</td>
                                        <td></td>
                                        </tr>
                                    </tbody>
                                </table>
                            </div>
                        </div>
                    </div> 
                </div>
            </div>
        )
    }
}
