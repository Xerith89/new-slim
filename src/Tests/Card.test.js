import React from 'react'
import {shallow, configure,} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Card,CardTitle, CardBody} from '../Components/Core'

configure({adapter: new Adapter()});

describe('Card Component Behaves As Expected', () => {
    test('Card should render to DOM', () => {
        let wrapper = shallow(<Card />);
        expect(wrapper.html).toMatchSnapshot();
    });

    test('Card should render to DOM with CardTitle nested', () => {
        let wrapper = shallow(<Card> <CardTitle title="My Card" /> </Card>);
        expect(wrapper.html).toMatchSnapshot();
    });

    test('Card should render to DOM with full nesting', () => {
        let wrapper = shallow(<Card> <CardTitle title="My Card" /><CardBody /> </Card>);
        expect(wrapper.html).toMatchSnapshot();
    });

});
