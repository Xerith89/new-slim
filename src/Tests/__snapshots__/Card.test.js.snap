// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Behaves As Expected Card should render to DOM 1`] = `[Function]`;

exports[`Card Component Behaves As Expected Card should render to DOM with CardTitle nested 1`] = `[Function]`;

exports[`Card Component Behaves As Expected Card should render to DOM with full nesting 1`] = `[Function]`;
